### PR TITLE
delete items #12

### LINKED ISSUE #12: As an admin, I want to be able to delete items
- [ ] Implement Logic for Deleting Items from Database
      - Should they be permanently deleted?
- [ ] Implement Item Delete Endpoint Class
- [ ] Implement API Endpoint for Deleting Items

### PR DESCRIPTION
**Pull request by MatteoGuarnaccia5**
## Description
Implemented API endpoint that allows users to delete items by supplying an ID. Invalid and non-existent IDs raises the appropriate error

## Testing instructions

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] test with non-existent ID
- [ ] test with invalid ID

## Agile board tracking
closes #12

### CODE REVIEW

File: inventory_management_system_api/routers/v1/item.py [hunk 1]
```diff
@@ -51,6 +51,26 @@ def create_item(item: ItemPostRequestSchema, item_service: ItemService = Depends
...
 
+@router.delete(
+    path="/{item_id}",
+    summary="Delete an item by ID",
+    response_description="Item deleted successfully",
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+def delete_item(
+    item_id: str = Path(description="The ID of the item to delete"),
+    item_service: ItemService = Depends(),
```
Comment (joelvdavies) [inline inventory_management_system_api/routers/v1/item.py:62]: Could these use `Annotated` like the others for #117.

**Review by joelvdavies — CHANGES_REQUESTED** [review]
Just one minor thing, could you also resolve the comment from https://github.com/ral-facilities/inventory-management-system-api/pull/133#discussion_r1439369322 here please.

**Review by joelvdavies — APPROVED** [review]

File: inventory_management_system_api/routers/v1/item.py [hunk 2]
```diff
@@ -54,6 +54,26 @@ def create_item(
...
 
+@router.delete(
+    path="/{item_id}",
+    summary="Delete an item by ID",
+    response_description="Item deleted successfully",
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+def delete_item(
+    item_id: Annotated[str, Path(description="The ID of the item to delete")],
+    item_service: Annotated[ItemService, Depends()],
```
Comment (VKTB) [inline inventory_management_system_api/routers/v1/item.py:65]:
```suggestion
    item_service: Annotated[ItemService, Depends(ItemService)],
```

File: test/unit/services/test_catalogue_category.py [hunk 3]
```diff
@@ -698,6 +698,9 @@ def test_update_change_catalogue_item_properties_when_has_children(
     )
     # pylint: enable=duplicate-code
 
+    # Mock so child elements found
+    catalogue_category_repository_mock.has_child_elements.return_value = True
```
Comment (VKTB) [inline test/unit/services/test_catalogue_category.py:702]: Why is this needed?

File: inventory_management_system_api/routers/v1/item.py (hunk 2 above)
Comment (joelvdavies) [reply inventory_management_system_api/routers/v1/item.py:65]: There is one like this in `get_item` as well.

File: test/unit/services/test_catalogue_category.py (hunk 3 above)
Comment (joelvdavies) [reply test/unit/services/test_catalogue_category.py:702]: This was removed in #123, see comment here https://github.com/ral-facilities/inventory-management-system-api/pull/133#discussion_r1439369322.

Comment (VKTB) [reply test/unit/services/test_catalogue_category.py:702]: Oh I see, I suggest that `develop` gets merged into this branch then. Ping @MatteoGuarnaccia5.

Comment (joelvdavies) [reply test/unit/services/test_catalogue_category.py:702]: @VKTB Ah sorry no I meant it was removed by accident, it's not in develop as that PR was merged after Joshua approved that PR, but before I had approved it, so I asked it to be re-added here as I had requested in that PR.

Comment (VKTB) [reply test/unit/services/test_catalogue_category.py:702]: @joelvdavies Ah sorry, I now get what you meant.

**Review by joelvdavies — APPROVED** [review]

**Review by VKTB — APPROVED** [review]
